src/byrow.jl

```diff
@@ -58,18 +62,19 @@ end
 
 Act on a DataFrame row-by-row.
 
+Changes to the rows do not affect `d` but instead the new data frame returned by
```

Comment:
I think it would be best to port the `@byrow!` documentation over to `@byrow` and leave the `@byrow!` docstring as. 

```
Deprecated version of `@byrow`, see: [`@byrow`](@ref)
```

Reply:
Fixed.